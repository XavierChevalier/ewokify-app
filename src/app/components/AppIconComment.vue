<script setup lang="ts">
  import { mdiSpeechOutline } from '@/assets/images/icons/MdiExtra'
  import Icon from '@/app/components/AppIcon.vue'
  import { isPropertyValid } from '@/app/tools/component-properties/PropertyValidator'
  import { number } from 'yup'

  defineProps({
    showCounter: {
      type: Boolean,
      default: false,
    },
    counter: {
      type: Number,
      default: 0,
      validator: isPropertyValid(number().integer().min(0)),
    },
  })
</script>

<template>
  <span class="flex text-neutral">
    <Icon :path="mdiSpeechOutline" />
    <span v-if="showCounter" class="ml-2">{{ counter }}</span>
  </span>
</template>
