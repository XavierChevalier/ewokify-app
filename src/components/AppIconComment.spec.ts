import { generateStorybookSnapshotTests } from '@/tests/tools/StorybookSnapshot'
import AppIconComment from './AppIconComment.vue'
import {
  IconCommentWithCounter,
  JustIconComment,
} from '@/components/AppIconComment.stories'
import { ComponentTestsGenerator } from '@/tests/tests-generators/ComponentTestsGenerator'

describe('AppIconComment', () => {
  generateStorybookSnapshotTests({ JustIconComment, IconCommentWithCounter })

  new ComponentTestsGenerator(AppIconComment)
    .itShouldBeDefined()
    .property<number>('counter')
    .itShouldBeDefined()
    .itShouldHaveAValidatorFunction()
    .itShouldSuccessWith(0)
    .itShouldSuccessWith(10)
    .itShouldFailWith('negative number', -1)
})
