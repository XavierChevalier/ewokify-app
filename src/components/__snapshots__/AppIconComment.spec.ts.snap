// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIconComment [snapshot] should render the "IconCommentWithCounter" component correctly 1`] = `
<span
  class="flex text-neutral"
>
  <anonymous-stub
    path="[object Object]"
    size="base"
  />
  <span
    class="ml-2"
  >
    10
  </span>
</span>
`;

exports[`AppIconComment [snapshot] should render the "JustIconComment" component correctly 1`] = `
<span
  class="flex text-neutral"
>
  <anonymous-stub
    path="[object Object]"
    size="base"
  />
  <!--v-if-->
</span>
`;
